File: server/internal/model/plot_structure.go
// server/internal/model/plot_structure.go
package model

import "time"

// PlotStructureTemplate 剧情结构模板
type PlotStructureTemplate struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Name        string    `gorm:"size:100;not null" json:"name"`           // 三幕式、英雄之旅...
	Category    string    `gorm:"size:30;not null" json:"category"`        // classic/web_novel/suspense/romance/custom
	Description string    `gorm:"type:text" json:"description"`
	Structure   string    `gorm:"type:text;not null" json:"structure"`     // JSON: [{phase, name, description, ratio, beats}]
	IsSystem    bool      `gorm:"not null;default:false" json:"is_system"`
	UserID      uint      `gorm:"index;default:0" json:"user_id"`         // 0=系统预置
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: web/src/api/plot_structure.ts
// web/src/api/plot_structure.ts
import request from './request'

// ========== 类型定义 ==========

export interface PlotPhase {
  phase: number
  name: string
  description: string
  ratio: number
  beats: string[]
}

export interface PlotStructureTemplate {
  id: number
  name: string
  category: string
  description: string
  structure: string // JSON string of PlotPhase[]
  is_system: boolean
  user_id: number
  created_at: string
  updated_at: string
}

// ========== API ==========

export const plotStructureApi = {
  list: () => request.get('/plot-templates'),
  get: (id: number) => request.get(`/plot-templates/${id}`),
  create: (data: { name: string; description?: string; structure: string }) =>
    request.post('/plot-templates', data),
  aiGenerate: (data: { description: string; model_name?: string }) =>
    request.post('/plot-templates/ai-generate', data),
  update: (id: number, data: { name?: string; description?: string; structure?: string }) =>
    request.put(`/plot-templates/${id}`, data),
  delete: (id: number) => request.delete(`/plot-templates/${id}`),
}
